src/qtism/common/datatypes/files/FileSystemFileManager.php

```diff
@@ -93,12 +93,13 @@ public function createFromFile($path, $mimeType, $filename = '')
      * @param string $data The binary data of the FileSystemFile object to be created.
      * @param string $mimeType A mime-type.
      * @param string $filename A file name e.g. "myfile.txt".
+     * @param string|null $path A path for file provided externally
      * @return FileSystemFile
      * @throws FileManagerException
      */
-    public function createFromData($data, $mimeType, $filename = '')
+    public function createFromData($data, $mimeType, $filename = '', $path = null)
     {
-        $destination = $this->buildDestination();
+        $destination = $path ?: $this->buildDestination();
```

Comment:
as I understand the ticket correctly you need only filename, why you pass full file pass?

Reply:
this place is not actual solution for ticket issue.
changes in this class mostly just to support extended method signature in interface

partially provided in other files if this pr and in bundle-qti https://github.com/oat-sa/bundle-qti/pull/60